identity-service/src/notifications/sendNotificationEmails.js

```diff
@@ -188,122 +199,112 @@ async function processEmailNotifications (expressApp, audiusLibs) {
     // All users with notifications, including announcements
     let allUsersWithUnseenNotifications = [...pendingNotificationUsers]
 
-    let userInfo = await models.User.findAll({
+    const userInfo = await models.User.findAll({
       where: {
         blockchainUserId: {
           [models.Sequelize.Op.in]: allUsersWithUnseenNotifications
         }
       }
     })
 
-    const timeBeforeUserEmailLoop = Date.now()
-    logger.info(`processEmailNotifications | time before looping over users to send notification email | ${timeBeforeUserEmailLoop} | ${userInfo.length} users`)
-    // For every user with pending notifications, check if they are in the right timezone
-    for (let userToEmail of userInfo) {
-      let userEmail = userToEmail.email
-      let userId = userToEmail.blockchainUserId
-      let timezone = userToEmail.timezone
-      if (!timezone) {
-        timezone = 'America/Los_Angeles'
-      }
-      let userSettings = await models.UserNotificationSettings.findOrCreate(
-        { where: { userId } }
-      )
-      let frequency = userSettings[0].emailFrequency
-      if (frequency === 'off') {
-        logger.info(`processEmailNotifications | Bypassing email for user ${userId}`)
-        continue
+    const userNotificationSettings = await models.UserNotificationSettings.findAll({
+      where: {
+        userId: {
+          [models.Sequelize.Op.in]: allUsersWithUnseenNotifications
+        }
       }
-      let currentUtcTime = moment.utc()
-      let userTime = currentUtcTime.tz(timezone)
-      let startOfUserDay = userTime.clone().startOf('day')
-      let difference = moment.duration(userTime.diff(startOfUserDay)).asHours()
+    })
 
-      let latestUserEmail = await models.NotificationEmail.findOne({
-        where: {
-          userId
-        },
-        order: [['timestamp', 'DESC']]
-      })
-      let lastSentTimestamp
-      if (latestUserEmail) {
-        lastSentTimestamp = moment(latestUserEmail.timestamp)
-      } else {
-        lastSentTimestamp = moment(0) // EPOCH
-      }
+    const userFrequencyMapping = userNotificationSettings.reduce((acc, setting) => {
+      acc[setting.userId] = setting.emailFrequency
+      return acc
+    }, {})
 
-      // If the user is in live email mode, just send them right away
-      if (frequency === 'live') {
+    const timeBeforeUserEmailLoop = Date.now()
+    logger.info(`processEmailNotifications | time before looping over users to send notification email | ${timeBeforeUserEmailLoop} | ${userInfo.length} users`)
+
+    const currentUtcTime = moment.utc()
+    const results = await Promise.all(userInfo.map(async (user) => {
```

Comment:
should we batch this promise.all to a smaller number? i'm concerned if we make queries like for example on line 240 if there's 100 queries made concurrently outside a transaction so it takes up all the db clients we have